torch/distributed/_tensor/api.py

```diff
@@ -459,8 +460,10 @@ def distribute_tensor(
             )
 
     assert local_tensor is not None, "distributing a tensor should not be None"
+    # detach the local tensor passed to DTensor since after the construction
+    # of DTensor, autograd would work on top of DTensor instead of local tensor
     return DTensor(
-        local_tensor,
+        local_tensor.detach(),
```

Comment:
I'm not 100% familiar with the semantic of `distribute_tensor()` but the current implementation implies that you get a DTensor with no history out of it, even if the input did require gradients. Is that accurate?

Reply:
Oh yes this is sth I would like to discuss, `distribute_tensor` is usually used to generate a DTensor and replacing it with nn.Parameters (that being said, I think `distribute_tensor` mostly being used to create leaf DTensors), so I think `detach` call is proper here? because we pass in `tensor.requires_grad` instead of `local_tensor`, it would still become leaf of the autograd graph.

I was thinking that if we create a view here, then the gradient would flow back to the original tensor (which is pre-sharding), this is sth we don't want because we want the gradient to stay sharded (`.grad` should be a sharded DTensor instead).

Reply:
> I was thinking that if we create a view here, then the gradient would flow back to the original tensor

It would if local_tensor ever get involved in the graph indeed.

If the semantic of this function is to create a leaf, then detach() is the right move I think yes.

Reply:
Thanks for confirming! Yeah `distribute_tensor` is used to create leaf DTensors, so let's keep the `detach` callsite here, and I removed other callsites to use `view_as` instead as we discussed :)